data-model/src/main/java/io/micronaut/data/model/query/builder/sql/Dialect.java

```diff
@@ -79,12 +79,16 @@ public final boolean allowBatch() {
     }
 
     /**
-     * Determines whether
+     * Returns compatible dialect dataype.
      * @param type the type
-     * @return True if a string UUID is required
-     * @since 1.1.3
+     * @return The dialect compatible DataType
+     * @since 2.0.1
      */
-    public final boolean requiresStringUUID(@NonNull DataType type) {
-        return type == DataType.UUID && this.stringUUID;
+    public final DataType getDataType(@NonNull DataType type) {
+        if (type == DataType.UUID && this.stringUUID) {
+            return DataType.STRING;
+        } else {
+            return type;
+        }
```

Comment:
I think that promoting Dialect into a class with the responsability of handling all of the datatype and sql keywords needs would help unclutter the code in some places.
for example this code could be simplified by delgating to the dialect this kind of handling (no need for switch statements)
https://github.com/micronaut-projects/micronaut-data/blob/894f8c1795a567c512c183ddc5b50f0aa1408633/data-model/src/main/java/io/micronaut/data/model/query/builder/sql/SqlQueryBuilder.java#L397-L463